packages/drivers/odsp-driver/src/odspDriverUrlResolverForShareLink.ts

```diff
@@ -230,3 +235,27 @@ export class OdspDriverUrlResolverForShareLink implements IUrlResolver {
         return encodeOdspFluidDataStoreLocator(locator);
     }
 }
+
+/**
+ * Utility to resolve urls like the ones created by createOdspUrl which is driver inner
+ * url format and the ones which have things like driveId, siteId, itemId etc encoded in nav param.
+ * @param request - Request to be resolved.
+ */
+export async function resolveNavParamUrlUtil(request: IRequest): Promise<IOdspResolvedUrl> {
+    const requestToBeResolved = { headers: request.headers, url: request.url };
+    try {
+        const url = new URL(request.url);
+        const odspFluidInfo = getLocatorFromOdspUrl(url);
```

Comment:
I'm not familiar enough with the sceanrio, but I wonder if the scenarios that currently need to do this could just use getLocatorFromOdspUrl directly as a utility. Do we need the rest of the code here?

Reply:
@AndreiZe - do you know if the rest of the code here is necessary

Reply:
If the only utility it just to get the drive and itemid, then we don't need this. getLocatorFromOdspUrl() should be enough.

Reply:
@AndreiZe

Reply:
I removed this utility and added comments.